scripts/osfstorage/files_audit.py

```diff
@@ -0,0 +1,184 @@
+#!/usr/bin/env python
+# encoding: utf-8
+
+"""Verify that all OSF Storage files have Glacier backups and parity files,
+creating any missing backups.
+
+TODO: Add check against Glacier inventory
+Note: Must have par2 installed to run
+"""
+
+import os
+import logging
+
+import pyrax
+from boto.glacier.layer2 import Layer2
+
+from website.addons.osfstorage import model
+
+from scripts.osfstorage import utils as script_utils
+from scripts.osfstorage import settings as storage_settings
+
+
+container_primary = None
+container_parity = None
+vault = None
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def download_from_cloudfiles(version):
+    path = os.path.join(storage_settings.AUDIT_TEMP_PATH, version.location['object'])
+    if os.path.exists(path):
+        return
+    obj = container_primary.get_object(version.location['object'])
+    obj.download(storage_settings.AUDIT_TEMP_PATH)
+    return path
+
+
+def delete_temp_file(version):
+    path = os.path.join(storage_settings.AUDIT_TEMP_PATH, version.location['object'])
+    try:
+        os.remove(path)
+    except OSError:
+        pass
+
+
+def ensure_glacier(version, dry_run):
+    if version.metadata.get('archive'):
+        return
+    logger.info('Glacier archive for version {0} not found'.format(version._id))
+    if dry_run:
+        return
+    download_from_cloudfiles(version)
+    file_path = os.path.join(storage_settings.AUDIT_TEMP_PATH, version.location['object'])
+    glacier_id = vault.upload_archive(file_path, description=version.location['object'])
+    version.metadata['archive'] = glacier_id
+    version.save()
+
+
+def check_parity_files(version):
+    objs = list(container_parity.list_all(prefix=version.location['object']))
+    return len(objs) == 8
```

Comment:
Spot checking the existing par2 archive and testing locally with different file sizes it appeared 8 files (*.par, *.vol*.par2) was a consistent number for parity files created for each archive. However reviewing the logs of all recently created files I found this is actually not consistent, we'll change this Monday.